src/cpp/lpnamer/main/ProblemGeneration.cpp

```diff
@@ -111,6 +113,9 @@ std::filesystem::path ProblemGeneration::updateProblems() {
   auto weights_file = options_.WeightsFile();
   auto unnamed_problems = options_.UnnamedProblems();
 
+  SettingsReader settingsReader(xpansion_output_dir, logger.get());
```

Comment:
I don't understand how this works : you give xpansion output directory but then it does `check_file_exist` but you have only given the directory, I do not see where the path of the settings file is retrieved (probably I missed something in the code...)

Reply:
Indeed a mistake from me.